### PR TITLE
Jest tests for RD Entry form Elevate credit card component

### PR DESCRIPTION
**Pull request by npsp-reedestockton**
# Critical Changes

# Changes
Added jest tests for Enhanced Recurring Donations Elevate credit card component.

# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by npsp-reedestockton** [comment]
Tracking [W-8712407](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008wnMqIAI/view)

**Comment by npsp-reedestockton** [comment]
@andrewyu-salesforce @bdvorachek Would appreciate your review of this jest test PR.

File: src/lwc/rd2ElevateCreditCardForm/rd2ElevateCreditCardForm.html [hunk 1]
```diff
@@ -28,7 +27,7 @@ <h1 class='slds-text-body_regular' data-qa-locator='heading'>{labels.elevateWidg
                     </lightning-layout-item>
 
 
-                    <lightning-layout-item size="12" multiple-rows="true" class='slds-p-horizontal_small slds-p-top_small'>
```
Comment (force2b) [inline src/lwc/rd2ElevateCreditCardForm/rd2ElevateCreditCardForm.html:31]: I'm curious why the need to remove multiple-rows here, and if this necessitates a deeper QA role on this work item then if it were just jest tests?

Comment (npsp-reedestockton) [reply src/lwc/rd2ElevateCreditCardForm/rd2ElevateCreditCardForm.html:31]: multiple-rows is not supported on lightning-layout-item.  It was causing a parse error during jest test execution.  It should never have been there.

Comment (force2b) [reply src/lwc/rd2ElevateCreditCardForm/rd2ElevateCreditCardForm.html:31]: Thanks

**Review by force2b — APPROVED** [review]

File: src/lwc/rd2ElevateCreditCardForm/__tests__/rd2ElevateCreditCardForm.test.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        document.body.appendChild(element);
+
+        return flushPromises()
+            .then(() => {
+                expect(isLoadingSpinner(element)).toBeTruthy();
+            });
+    });
+
+    it('should allow disabling and enabling of widget', async () => {
+        let element = createWidget();
```
Comment (bdvorachek) [inline src/lwc/rd2ElevateCreditCardForm/__tests__/rd2ElevateCreditCardForm.test.js:76]:
```suggestion
        const element = createWidget();
```

File: src/lwc/rd2ElevateCreditCardForm/__tests__/rd2ElevateCreditCardForm.test.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+);
+
+const GET_ORG_DOMAIN_ERROR = {
+    body: { message: ALERT_MESSAGE },
+    ok: false,
+    status: 400,
+    statusText: 'Bad Request'
+};
+
+function flushPromises() {
```
Comment (bdvorachek) [inline src/lwc/rd2ElevateCreditCardForm/__tests__/rd2ElevateCreditCardForm.test.js:35]: `flushPromises` is defined in [jest.setup.js](https://github.com/SalesforceFoundation/NPSP/blob/master/jest.setup.js). Things put on the global jest object don't need to get defined in the context of an individual testfile.

File: src/lwc/rd2ElevateCreditCardForm/__tests__/rd2ElevateCreditCardForm.test.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
+import { createElement } from 'lwc';
+import rd2ElevateCreditCardForm from 'c/rd2ElevateCreditCardForm';
+import { registerSa11yMatcher } from "@sa11y/jest";
```
Comment (bdvorachek) [inline src/lwc/rd2ElevateCreditCardForm/__tests__/rd2ElevateCreditCardForm.test.js:3]:
```suggestion
import { registerSa11yMatcher } from '@sa11y/jest';
```

**Review by bdvorachek — CHANGES_REQUESTED** [review]

**Comment by npsp-reedestockton** [comment]
Thanks, @bdvorachek.  Requested changes have been applied.

**Review by bdvorachek — APPROVED** [review]

**Pull request merged**